packages/react-dom/src/client/ReactDOMComponentTree.js

```diff
@@ -40,17 +37,13 @@ export function getClosestInstanceFromNode(node) {
     }
   }
 
-  let closest;
   let inst = node[internalInstanceKey];
   if (inst.tag === HostComponent || inst.tag === HostText) {
     // In Fiber, this will always be the deepest root.
     return inst;
   }
-  for (; node && (inst = node[internalInstanceKey]); node = parents.pop()) {
-    closest = inst;
-  }
 
-  return closest;
```

Comment:
Does it make sense to return `null` for invalid fiber types? Maybe it should throw if it's not a `HostComponent` or `HostText`? I'm not sure if there are any situations where this should be called with any other type of node.

Reply:
that's a good question. I think your probably right, however i was trying to keep with the spirit (for lack of a better term) of the method by failing softly.

I do think tho that a situation where its not a component or text is kind of a different sort of error and maybe should fail hard. Are roots or portals cached on elements? Presumable you should hit a component before that tho